src/ui/components/shared/WorkspaceSettingsModal/GeneralSettings.tsx

```diff
@@ -0,0 +1,69 @@
+import classNames from "classnames";
+import React, { useEffect, useRef, useState } from "react";
+import hooks from "ui/hooks";
+
+const Label = ({ className, children }: { className?: string; children: React.ReactNode }) => {
+  return (
+    <div className={classNames(className, "w-4/12 flex-shrink-0")}>
+      <label>{children}</label>
+    </div>
+  );
+};
+
+const Input = ({ children }: { children: React.ReactNode }) => {
+  return <div className="w-8/12">{children}</div>;
+};
+
+const Row = ({ children }: { children: React.ReactNode }) => {
+  return <div className="flex items-center">{children}</div>;
+};
+
+const GeneralSettings = ({ workspaceId }: { workspaceId: string }) => {
+  const { workspace } = hooks.useGetWorkspace(workspaceId!);
+  const updateWorkspaceSettings = hooks.useUpdateWorkspaceSettings();
+  const [name, setName] = useState(workspace?.name);
+  const ref = useRef<NodeJS.Timeout | undefined>();
+
+  useEffect(() => {
+    if (ref.current) {
+      clearTimeout(ref.current);
+    }
+
+    ref.current = setTimeout(() => {
+      updateWorkspaceSettings({
```

Comment:
If I'm reading this right, when combined with `useState(workspace?.name)`, it's possible to blow away the workspace's name and replace it with undefined here.

Reply:
I considered this too but I didn't experience this in practice since `workspace` was already in apollo cache by the time you hit this. I do think we should be more defensive about it though so i'll touch it up tomorrow.